src/evict/evict_page.c

```diff
@@ -677,6 +677,104 @@ __evict_child_check(WT_SESSION_IMPL *session, WT_REF *parent)
     return (0);
 }
 
+/*
+ * __evict_review_obsolete_time_window --
+ *     Check whether the ref has obsolete time window information and mark it for dirty eviction to
+ *     remove those obsolete data. An exclusive lock on the page has already been obtained by the
+ *     caller.
+ */
+static int
+__evict_review_obsolete_time_window(WT_SESSION_IMPL *session, WT_REF *ref)
```

Comment:
This can only happen in clean eviction?

Reply:
As part of the dirty eviction, we already perform the reconciliation and the obsolete time window is cleared. So this function changes the clean page into the dirty page.